### ISSUE TITLE
Tencrop on test images & preprocessing imbalanced data

### ISSUE THREAD

**Issue by han-liu** [arnoweng/CheXNet#24]
Hi arnoweng,

I have two following questions in this project.
1):
I'm wondering why to apply tencrop technique on testing images. I thought data augmentation techniques should only be applied on training set in order to add diversity of training images while test images should be kept unchanged so the testing results can be compared with others. If you use tencrop on testing images, you are technically using a different testing set, right?
2):
Do you think if it is necessary to pre-process the imbalanced data? I noticed there is a HUGE imbalance between the sample number of hernia and other diseases (~200 images vs 10000 e.g. infiltration), and thus different testing set would result in really different aucroc on at least Hernia. e.g. If my test set accidentally include only just 10 Hernia images, I guess the aucroc score of Hernia in this test set would be really high like 93%. In contrast, if there are around 150 Hernia in test set, the aucroc score would be low. 
Thanks a lot!

**Comment by singhsegv** [comment]
Did you ever get the answer for this?

**Comment by han-liu** [comment]
Not yet. @singhsegv

**Comment by quockhangdev** [comment]
For Q1, I think it is TTA @han-liu 
Source: https://docs.ultralytics.com/yolov5/tutorials/test_time_augmentation/#inference-with-tta
